forem/forem
# Move profile initialization to after_initialize hook

## What type of PR is this? (check all applicable)


## Description

This PR moves profile initialization from a custom hack to a proper Rails solution using `config.after_initialize`. Thanks to @rhymes for pushing me to give this another try, I have no idea why it didn't work the last time I tried exactly the same 🤷 (well, I have some theories, but it really doesn't matter).

## Related Tickets & Documents

#6994 

## QA Instructions, Screenshots, Recordings

1. Ensure sure all specs pass.
2. Drop your database and make sure you can run `bin/setup` to completion.
3. Ensure profiles are ready to use right after app boot, e.g. by starting a Rails console and verifying that the number of attributes is correct without explicitly calling `Profile.refresh_attribute!`:
    ```ruby
    Profile.attributes.count == ProfileField.count.succ
    ```
## Added tests?

- [X] No, and this is why: plenty of existing tests that should still pass.

## Added to documentation?
